gdrive

```diff
@@ -284,5 +234,77 @@ def main():
     elif func == 'list':
         list_files(service, sys.argv[2])
 
+
+class ConflatedChannel:
+    def __init__(self):
+        self.__channel = []
+        self.__open = True
+
+    def __str__(self):
+        return str(self.__channel)
+
+    def send(self, value):
+        self.__channel.clear()
+        self.__channel.append(value)
+
+    def pop(self):
+        try:
+            return self.__channel.pop()
+        except:
+            return None
+
+    def isOpen(self):
+        return self.__open
+
+    def close(self):
+        self.__open = False
+
+class Logger():
+    def __init__(self, operation):
+        self.__channel = ConflatedChannel()
+        self.__operation = operation
+        self.__start_time = time.time()
+        self.__worker = threading.Thread(target=self.work, daemon=True)
+        self.__worker.start()
+
+    def send(self, status):
+        self.__channel.send(status)
+
+    def close(self):
+        self.__channel.close()
+        self.__worker.join()
+
+    def work(self):
```

Comment:
this should never be called outside the scope of the `Logger`, so make it private (`__work()`)

Reply:
Agreed, done.